designdocs/schemas_protocols_intents.md

```diff
@@ -0,0 +1,366 @@
+- **Schema** - A `Schema` represents a particular data format, along with additional metadata about its location, etc
+
+## 4. **Protocols** <a id="section_4"></a>
+
+In order to provide organizational control for schemas and intents, we introduce the concept of **Protocols**.
+
+A Protocol (e.g., `dsnp`, `bsky`) serves as a root authority under which Schemas and Intents are registered.
+Protocols allow for Intents and Schemas with similar meaning or structure, but different purposes, to be delegated
+in a scoped manner. For example, there could be both a 'dsnp.broadcast' and a 'bsky.broadcast' Intent; both have similar
+meaning ('broadcast a message'), but the Protocol differentiates between broadcasting a DSNP message vs. an AT
+Protocol (ie, Bluesky) message.
+
+### 📛 Protocol Structure
+
+Each protocol is a unique identifier, typically human-readable (e.g., `"dsnp"`), mapped to an on-chain `ProtocolId` (
+e.g., `ProtocolId = 2`).
+
+- Every Intent must be published under a protocol: `protocol.intent`
+
+### ⚙️ Protocol Governance
+
+- **Creation of a new protocol** must be approved via governance.
+- Protocols are immutable once registered
+
+## 5. **Intents** <a id="section_5"></a>
+
+Intents are not just organizational tools—they enable meaningful delegation. In the new model, an Intent is a
+first-class on-chain entity. It can represent a single action or permission that may be delegated; it can optionally
+reference a list of Schemas that comprise minor version updates of a particular data format--in the latter case, the
+implied permission is `schema.Write`. Every published Schema is assigned to an Intent, and new versions of a schema
+simply add new schema IDs to the Intent's version history.
+
+As mentioned above, an `Intent` may reference a list of `Schemas` that comprise the sequential _versions_ to which a
+delegation to the Intent grants _write_ access. However, it is also possible to create an Intent that does not reference
+_any_ Schemas; such an Intent may be used as a way to record arbitrary permissions on-chain. These permission indicators
+may be used by off-chain applications (or future on-chain facilities). For example, a wallet application may query an
+on-chain Intent to determine whether an application should be allowed access to a user's graph encryption key.
+
+### 📌 Intent Metadata Structure
+
+- `IntentId`: A unique numeric identifier for runtime efficiency
+- `ProtocolId`: Which Protocol owns this Intent
+- `IntentId`: Compact numeric ID used on-chain
+- `Name`: Human-readable name (e.g., `broadcast`)
+- `Description`: Optional developer-facing documentation
+- `Link`: Optional link to spec or docs
+- `RegisteredBy`: MSA (ProviderId) that registered the intent
+- `RegisteredAt`: Block number
+- `Versions`: [optional] Ordered list of `SchemaId`s (e.g., `[7, 15, 27]`, for versions 0..2)
+
+### 🧬 Versioning Rules
```

Comment:
What if we dropped versions and instead just did schema deprecation tagging?

Reply:
Not sure I understand. Elaborate?

Reply:
I have a bit more here: https://github.com/frequency-chain/frequency/pull/2312#discussion_r2219482941

Versions imply a semantic that we might not want to imply. Also some protocols might use other ways of versioning than int incrementing.

Effectively with Intents, we can just say "here are all the schemas for this intent and these ones are flagged as deprecated"

The schema naming we do could also be updated. With named Protocols and Intents, the only reason to name schemas is for ease of developer integration across mainnet and testnets.

Reply:
Tracking here: #2534